src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/PrisonerContactRegistryClient.kt

```diff
@@ -57,4 +68,60 @@ class PrisonerContactRegistryClient(
     uriBuilder.queryParam("withAddress", withAddress)
     return uriBuilder
   }
+
+  fun getBannedRestrictionDateRange(prisonerId: String, visitors: List<Long>, prisonDateRange: DateRange): DateRange {
+    val uri = "/prisoners/$prisonerId/approved/social/contacts/restrictions/banned/dateRange"
+
+    return webClient.get()
+      .uri(uri) {
+        getBannedRestrictionDateRangeParams(visitors, prisonDateRange, it)
+      }
+      .retrieve()
+      .bodyToMono<DateRange>()
+      .onErrorResume {
+          e ->
+        if (!isNotFoundError(e)) {
+          LOG.error("getBannedRestrictionDateRage Failed get request $uri")
+          Mono.error(e)
+        } else {
+          val message = "getBannedRestrictionDateRage NOT FOUND get request $uri"
+          LOG.error("getBannedRestrictionDateRage NOT FOUND get request $uri")
+          Mono.error(DateRangeNotFoundException(message, e))
+        }
+      }
+      .blockOptional(apiTimeout).get()
```

Comment:
get or OrElseThrow? Other clients got OrElseThrow.

Reply:
My bet is it will never get to this point to throw an exception at this point (Null pointer exception), as the service will always throw something or return a valid object and any time out will throw an exception. I think this is a Mono block issue.